src/saml2/response.py

```diff
@@ -722,11 +722,12 @@ def _bearer_confirmed(self, data):
         return True
 
     def _holder_of_key_confirmed(self, data):
-        if not data:
+        if not data or not data.key_info:
             return False
 
         has_keyinfo = False
-        for element in extension_elements_to_elements(data,
+        key_info = data.key_info or ()
+        for element in extension_elements_to_elements(key_info,
                                                       [samlp, saml, xenc, ds]):
```

Comment:
The needed change here is

```py
for element in extension_elements_to_elements(
    data.extension_elements, [samlp, saml, xenc, ds]
):
    ...
```

`extension_elements_to_elements()` will map the extension elements to objects (ie, in this case an array with an element of `ds.KeyInfo` will be returned)

Reply:
Thanks. I changed the code to use `extension_elements` instead.